torch/distributed/_sharding_spec/api.py

```diff
@@ -41,6 +41,34 @@ class ShardingSpec(PlacementSpec):
     pass
 
 
+@dataclass
+class ReshardingSpec(object):
+    """
+    This is a wrapper of PlacementSpec that defines the new placement as being
```

Comment:
There is no PlacementSpec here, it is a wrapper of ShardingSpec.

Reply:
Sure. I first thought we have a PlacementSpec inside the ShardingSpec. Let me change it.